bot/cogs/help_channels.py

```diff
@@ -713,6 +717,32 @@ async def on_message(self, message: discord.Message) -> None:
         # be put in the queue.
         await self.move_to_available()
 
+    @commands.Cog.listener()
+    async def on_message_delete(self, msg: discord.Message) -> None:
+        """Reschedule dormant when help channel is empty."""
+        if not self.is_in_category(msg.channel, constants.Categories.help_in_use) or not self.is_empty(msg.channel):
+            return
+
+        log.info(f"Claimant of #{msg.channel} ({msg.author}) deleted message, channel is empty now. Rescheduling task.")
+
+        # Cancel existing dormant task before scheduling new.
+        self.cancel_task(msg.channel.id)
+
+        task = TaskData(constants.HelpChannels.deleted_idle_minutes * 60, self.move_idle_channel(msg.channel))
+        self.schedule_task(msg.channel.id, task)
+
+    async def is_empty(self, channel: discord.TextChannel) -> bool:
```

Comment:
Take a look at `is_dormant_message`. It strips the messages and also checks that the author is this bot and not some other bot, but basically does the same thing as `is_empty`. You could make `is_dormant_message` a generic function that can compare to any message and then call that in `is_empty`.